### DISCUSSION TITLE
[Demande d'avis à l'écosystème] Faut-il fusionner les deux PractitionerRole ?

### DISCUSSION THREAD

**Discussion by nriss** [Interop-Sante/hl7.fhir.fr.core#165] [Ideas]
# Historique

FrCore contient deux profils PractitionerRole, un pour la situation d'exercice, l'autre pour l'exercice professionnel. Il s'agit d'un héritage du modèle MOS.

Les travaux annuaire santé ont permis de changer cela pour être plus proche de la philosophie FHIR : Un PractitionerRole pour décrire la situation d'exercice d'un professionnel, un ou plusieurs Practitioner pour décrire la profession du praticien (ex : médecin + pharmacien), une ressource Person pour décrire l'identité civile de la personne.

Avant, pour un médecin-pharmacien, on aurait eu:
- Un PractitionerRole situation d'exercice
- Un PractitionerRole exercice professionnel (savoir faire, autorité d'enregistrement)
- Un Practitioner décrivant l'identité civile du médecin (diplôme)

Avec cette nouvelle version, nous avons :
- Un PractitionerRole situation d'exercice
- Un Practitioner par profession médicale (diplôme, savoir faire, autorité d'enregistrement)
- Une ressource Person pour décrire l'identité civile (nom, prénom, etc ...)

Cf : https://interop.esante.gouv.fr/ig/fhir/annuaire/#standards-utilis%C3%A9s

Essai côté FrCore : https://github.com/Interop-Sante/hl7.fhir.fr.core/pull/61
Issue associée : #82 

# FHIR 

La plupart des ressources FHIR ont deux attributs qui peuvent être confondus : 

* L'attribut `id` de cardinalité 0..1 correspondant à l'id technique (également appelé id logique) d'une ressource FHIR, assignée par un serveur, unique et ne changeant pas

Sources : 
https://www.hl7.org/fhir/R4/datatypes.html#id
![image](https://github.com/Interop-Sante/hl7.fhir.fr.core/assets/48218773/90be8f4a-2712-4423-85bf-d70a0030d22a)

![image](https://github.com/Interop-Sante/hl7.fhir.fr.core/assets/48218773/ab5e8a82-bb39-437a-99d9-9082ed1bc4aa)

https://www.hl7.org/fhir/R4/resource.html#id
![image](https://github.com/Interop-Sante/hl7.fhir.fr.core/assets/48218773/d614f69a-5fb0-43fb-9326-aceb5332f1f1)

* L'attribut `identifier`de cardinalité 0..* correspond à l'identifiant métier de la ressource (ex : RPPS, FINESS, ...)

Dans le cas du Practitioner qui peut avoir plusieurs ressources pour chaque spécialité (médecin, pharmacien, ...), le serveur pourra diffuser plusieurs Practitioner avec des id différents mais le même identifier RPPS.

# Demande d'avis

Pensez-vous qu'il faut fusionner les deux profils PractitionerRole pour avoir un unique ?
Cela permettrait de simplifier l'usage de ce profil, d'être plus proche de FHIR, et d'éviter les divergences entre FrCore et l'annuaire santé

Si nécessaire, un GT pourra être organisé pour en discuter.

**Comment by wylem-bars** [comment]
Pour moi c'est évident. Il faut fusionner les deux profils. C'est un sujet qui avait été remonté pour l'annuaire santé de notre côté et de la même façon, ce principe s'applique au FrCore.

**Comment by Monstermanu** [comment]
D'accord sur le principe.
Petite question formelle cependant (parce qu'opérationnellement il ne me semble pas que ça pose de problème), dans le cas des multi-professions, ça veut dire qu'on a plusieurs ressources Practitioner qui partage un même identifier. Est-ce que ce n'est pas problématique par rapport à la définition FHIR de identifier "A numeric or alphanumeric string that is associated with a single object or entity within a given system."?

**Reply by nriss** [reply]
C'est intéressant comme remarque! J'ai l'impression que ça ne pose pas de soucis, parce que quand on regarde dans practitioner, on a ces infos :

- La documentation indique qu'il peut y avoir plusieurs Practitioner pour chaque rôle

<img width="1254" alt="image" src="https://github.com/Interop-Sante/hl7.fhir.fr.core/assets/48218773/37417654-ae47-4744-adaa-e70330f1d26f">

- L'identifier de practitioner permet d'identifier la "personne comme agent", correspondant bien au RPPS

<img width="658" alt="image" src="https://github.com/Interop-Sante/hl7.fhir.fr.core/assets/48218773/996ed314-db99-4a80-b952-e17c1b3400c2">

J'ai pas l'impression que ça pose de soucis de mon côté

**Reply by Monstermanu** [reply]
Top. Merci de la réponse
